### PR TITLE
chore(hybrid-cloud): Adds created date to outboxes for better lag time calculation

### PR DESCRIPTION
**Pull request by GabeVillalobos**
Adds a new `date_added` field to outboxes for us to properly calculate lag time before processing an outbox event. This will help us with outbox alerting and monitoring if we have messages that fail to process for an extended period of time.

### CODE REVIEW

**Review by corps — APPROVED** [review]

**Review by mrduncan — COMMENTED** [review]
Should we add a `metrics.timing` calculation for this?

**Comment by corps** [comment]
> Should we add a `metrics.timing` calculation for this?

We can do that in a follow up.

File: src/sentry/migrations/0535_add_created_date_to_outbox_model.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    reverse_sql="""
+                    ALTER TABLE "sentry_regionoutbox" DROP COLUMN "created_date"
+                    """,
+                    hints={"tables": ["sentry_regionoutbox"]},
+                ),
+            ],
+            state_operations=[
+                migrations.AddField(
+                    model_name="controloutbox",
+                    name="created_date",
```
Comment (markstory) [inline src/sentry/migrations/0535_add_created_date_to_outbox_model.py:51]: Most of our models use `date_added` as the name for this column.

**Review by markstory — COMMENTED** [review]
Would be good to re-use the `date_added` convention we use in other models.

File: src/sentry/migrations/0535_add_created_date_to_outbox_model.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                    field=models.DateTimeField(default=django.utils.timezone.now, editable=False),
+                ),
+                migrations.AddField(
```
[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/migrations/0535_add_created_date_to_outbox_model.py:54]: django will complain about inconsistent model state until we add a migration there to update schema.

**Comment by GabeVillalobos** [comment]
> Should we add a `metrics.timing` calculation for this?

@mrduncan Yep! I'll be adding that as a follow-up commit after getting this in both sentry and getsentry.

**Pull request merged**